src/libstd/panicking.rs

```diff
@@ -149,6 +149,18 @@ impl<'a> PanicInfo<'a> {
         self.payload
     }
 
+    /// Returns the payload as a slice if it is a `&'static str` or `String`.
```

Comment:
Downcasting to `&AsRef<str>` would work for this signature, if that feature is ever implemented. But yeah you might actually want `&Display` instead.